### PR TITLE
adding condition to filter unattempted players who left clan

### PR DESCRIPTION
**Pull request by josephwccheng**

### CODE REVIEW

File: clash_royale_service.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def _get_players_remaining(self, participants: List[Any], clan_members: List[Any
         players_remaining = []
         
         for tag in tag_to_player_war:
-            if tag not in tag_to_player_clan and exlcude_not_in_clan:
+            if tag not in tag_to_player_clan and (exlcude_not_in_clan or tag_to_player_war[tag]["decksUsedToday"] == 0):
```
Comment (damoster) [inline clash_royale_service.py:215]: Can you add a test for this?

Comment (josephwccheng) [reply clash_royale_service.py:215]: added test case

**Review by damoster — APPROVED** [review]
LGTM

File: tests/test_clan_remaning_war_attacks.py [hunk 2]
```diff
@@ -35,14 +35,22 @@ def test_get_players_remaining(self):
...
             {"tag": "C", "lastSeen": "20221206T202149.000Z"}
         ]
-        assert len(service._get_players_remaining(participants, members_list)) == 2
+
+        players_remaining = service._get_players_remaining(participants, members_list)
+        # Includes player A,B, and E
+        assert len(players_remaining) == 3
+        players_remaining = service._get_players_remaining(participants, members_list, True)
+        # Only Includes player A and B as E is out of the clan (excluded from the third argument in_clan)
+        assert len(players_remaining) == 2
```
[the remark this answers is not in the document]
Comment (josephwccheng) [reply tests/test_clan_remaning_war_attacks.py:55]: done

**Pull request merged**